docs/selecting-a-datastore.md

```diff
@@ -0,0 +1,112 @@
+# Selecting a datastore
+
+SpiceDB ships with a number of [datastores](/internal/datastore), drivers used to store and retrieve the relationship data used for computing permissions.
+
+The currently available datastores are:
+- [Cockroach DB](#cockroach-db) - Best for large or scalable installations
+- [Postgres](#postgres) - Best for smaller or isolated applications
+- [memdb](#memdb) - For development and testing
+
+## Cockroach DB
+
+[Implementation](/internal/datastore/crdb)
+
+Datastore for storing and accessing data in a [Cockroach DB](https://github.com/cockroachdb/cockroach) cluster.
+
+- Best for large or scalable installations
+- Fully distributed and resilient to node failures
+- Queries are sharded over the breadth of the cluster
+
+Caveats:
+
+- Slightly less performant for writing vs Postgres (see [Overlay Strategies](#overlay-strategies) below)
```

Comment:
I think we need some actual benchmarks to make statements like this. CRDB has the overlap strategy, but Postgres has the entire transaction table that crdb doesn't need. It's not obvious to me which is more performant at smaller scales.

Reply:
I can remove for the moment, but I recall it being slightly slower